### PR TITLE
Add tests for mission and quests in LogicManager, AddressBookParser, AddressBook and ModelManager

### PR DESCRIPTION
**Pull request by wilinetan**

### CODE REVIEW

File: src/test/java/seedu/jarvis/logic/LogicManagerTest.java [hunk 1]
```diff
@@ -1,162 +1,172 @@
...
+    private void assertCommandFailure(String inputCommand, Class<? extends Throwable> expectedException,
+            String expectedMessage, Model expectedModel) {
+        assertThrows(expectedException, expectedMessage, () -> logic.execute(inputCommand));
+        assertEquals(expectedModel, model);
+    }
+
+    /**
+     * A stub class to throw an {@code IOException} when the save method is called.
+     */
+    private static class JsonAddressBookIoExceptionThrowingStub extends JsonAddressBookStorage {
```
Comment (ngzhenteng) [inline src/test/java/seedu/jarvis/logic/LogicManagerTest.java:162]: Nice use of a stub!

File: src/test/java/seedu/jarvis/logic/commands/ViewUngradedQuestCommandTest.java [hunk 2]
```diff
@@ -39,4 +45,14 @@ public void execute_emptyModel_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> viewUngradedQuestCommand.execute(emptyModel));
     }
 
+    @Test
+    public void execute_viewUngradedQuestCommand_questListFiltered() {
```
Comment (ngzhenteng) [inline src/test/java/seedu/jarvis/logic/commands/ViewUngradedQuestCommandTest.java:49]: Nice way of using typicalQuest objects.

File: src/test/java/seedu/jarvis/ui/MainWindowTest.java [hunk 3]
```diff
@@ -4,23 +4,43 @@
...
+    @TempDir
+    public Path temporaryFolder;
+
     // This model is used to execute commands
     private Model model;
 
+    private Logic logic;
+
     @BeforeAll
     public void setUp() {
```
Comment (ngzhenteng) [inline src/test/java/seedu/jarvis/ui/MainWindowTest.java]: Thank you for writing this, i will use it to write integration tests for MainWindow, LogicManager and the other classes involved from methods called in MainWindow.java

File: src/test/java/seedu/jarvis/logic/LogicManagerTest.java [hunk 4]
```diff
@@ -1,162 +1,172 @@
...
-//            super(filePath);
-//        }
-//
-//        @Override
-//        public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) throws IOException {
-//            throw DUMMY_IO_EXCEPTION;
-//        }
-//    }
-//}
+package seedu.jarvis.logic;
```
Comment (peironggg) [inline src/test/java/seedu/jarvis/logic/LogicManagerTest.java:1]: Thanks for making the LogicManagerTest work!

File: src/test/java/seedu/jarvis/logic/commands/ViewQuestDeadlineCommandTest.java [hunk 5]
```diff
@@ -39,4 +46,14 @@ public void execute_emptyModel_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> viewQuestDeadlineCommand.execute(emptyModel));
     }
 
+    @Test
```
Comment (peironggg) [inline src/test/java/seedu/jarvis/logic/commands/ViewQuestDeadlineCommandTest.java:49]: great use of abstraction for this testcase 👍

File: src/test/java/seedu/jarvis/logic/parser/AddressBookParserTest.java [hunk 6]
```diff
@@ -73,8 +96,43 @@ public void parseCommand_help() throws Exception {
     }
 
     @Test
-    public void parseCommand_viewAllStudents() throws Exception {
+    public void parseCommand_view() throws Exception {
+        // View all students
         assertTrue(parser.parseCommand(ViewAllStudentsCommand.SAMPLE_COMMAND) instanceof ViewAllStudentsCommand);
```
Comment (peironggg) [inline src/test/java/seedu/jarvis/logic/parser/AddressBookParserTest.java:101]: Real comprehensive test cases for AddressBookParser

File: src/test/java/seedu/jarvis/model/ModelManagerTest.java [hunk 7]
```diff
@@ -100,6 +107,53 @@ public void getFilteredPersonList_modifyList_throwsUnsupportedOperationException
         assertThrows(UnsupportedOperationException.class, () -> modelManager.getFilteredStudentList().remove(0));
     }
 
+    @Test
```
Comment (peironggg) [inline src/test/java/seedu/jarvis/model/ModelManagerTest.java:110]: Great start to ModelManager

**Review by ngzhenteng — COMMENTED** [review]
LGTM! thank you

**Pull request merged**